### ISSUE TITLE
Update vjs.Component.addChild to allow an index

### ISSUE THREAD

**Issue by erikyuzwa** [videojs/video.js#2534]
I'm trying to add "skip next" and "skip back" vjs.Buttons which are currently plugins that would be surrounding the Play button in the ControlBar Component.

the Component supports the addChild(child, [options]), but it would be nice to be able to inject a component at a specific index to help the re-arrangement in the DOM of controls that are rendered.

(my apologies if this is a FAQ)

**Comment by heff** [comment]
#2477 is actually related to this one. It would make children an array, so a plugins could just manually insert itself in the children array before init.

But also no, there isn't a way to specify that today. If you'd be interested in helping add the option, that'd be cool. I could see the third argument of addChild be the index where you want to splice the child in.

**Comment by erikyuzwa** [comment]
most excellent @heff thanks for the feedback. I'd love to take a shot.

**Comment by thijstriemstra** [comment]
I'd also be interested in this.

**Comment by thijstriemstra** [comment]
Maybe a new `addChildAt`?

**Comment by chemoish** [comment]
+1

**Comment by chemoish** [comment]
This is what I am currently doing.

```js
let control_bar = this.player.getChild('controlBar');

let chapter_button = control_bar.getChild('chaptersButton');

let chapter_thumbnail_menu_button = new ChapterThumbnailMenuButton();

// add component to end of control bar
control_bar.addChild(chapter_thumbnail_menu_button);

// move component
control_bar.el().insertBefore(chapter_thumbnail_menu_button.el(), chapter_button.el());
```

Hope this helps as an alternative.

**Comment by gkatsev** [comment]
In 5.0 we also have a separator element you can use to insert before or after to make your components available no the right-ish or left-ish side of the control bar.

**Comment by gkatsev** [comment]
This has landed by #2540 in 5.7!
